packages/hash/frontend/src/pages/[account-slug]/types/entity-type/[entity-type-id].page/definition-tab/property-list-card/shared/property-type-form/data-type-selector/custom-data-type-menu/array-data-type-menu.tsx

```diff
@@ -278,9 +278,14 @@ export const ArrayDataTypeMenu: FunctionComponent<ArrayDataTypeMenuProps> = ({
           onChange={(_evt, _data, reason, details) => {
             const typeId = details?.option;
             if (typeId) {
+              const allowMultiple =
+                expectedValuesOptions[typeId]?.allowMultiple;
               const defaultData = getDefaultData(typeId);
 
-              if (reason === "selectOption") {
+              if (
+                reason === "selectOption" ||
+                (reason === "removeOption" && allowMultiple)
+              ) {
```

Comment:
I'm not sure if I understand what's happening here. If an option is selected, or if an option is removed, but it's "allowMultiple", then we add an item to the list?

Reply:
Autocomplete toggles items by default, which is something we want for the basic data types (text, number, boolean). So if we select text and text was already selected, it is removed. But for objects and arrays we want to be able to add as many as we want, because they can be different from each other.